packages/happy-dom/src/query-selector/QuerySelector.ts

```diff
@@ -106,7 +106,14 @@ export default class QuerySelector {
 		const selector = new SelectorItem(selectorParts[0]);
 		const result = selector.match(<IElement>currentNode);
 
-		if ((targetNode === currentNode || !currentNode.parentElement) && !result.matches) {
+		if (result.matches && selectorParts.length === 1) {
+			return {
+				priorityWeight: priorityWeight + result.priorityWeight,
+				matches: true
+			};
+		}
+
+		if (!currentNode.parentElement || (targetNode === currentNode && !result.matches)) {
```

Comment:
Refactored slightly to always verify parentElement is not null, making it pass null checks in case you ever turn on strictNullChecks.

Reply:
Yes that would be good to turn on 🙂